htlcswitch/mailbox.go

```diff
@@ -314,6 +321,18 @@ func (m *memoryMailBox) signalUntilShutdown(cType courierType) {
 	}
 }
 
+// pktWithExpiry wraps an incoming packet and records the time at which it it
+// should be canceled from the mailbox. This will be used to detect if it gets
+// stuck in the mailbox and inform when to cancel back.
+type pktWithExpiry struct {
+	pkt    *htlcPacket
+	expiry time.Time
+}
+
+func (p *pktWithExpiry) deadline(clock clock.Clock) <-chan time.Time {
+	return clock.TickAfter(p.expiry.Sub(clock.Now()))
```

Comment:
Given that multiple packets can be outstanding, perhaps we should be using a single channel here where all the expiry messages are sent over?

Reply:
due to clock monotonicity the next packet we can expire is always at `addHead`, so we only need to handle one at a time